### PR TITLE
Support reading partials from memory

### PR DESCRIPTION
**Pull request by iffy**
I'd like to be able to make a single executable with embedded templates.

### CODE REVIEW

File: src/mustachepkg/values.nim [hunk 1]
```diff
@@ -24,13 +24,16 @@ type
...
 
 proc derive*(val: Value, c: Context): Context;
 
-proc newContext*(searchDirs = @["./"]): Context =
-  Context(values: initTable[string,Value](), searchDirs: searchDirs)
+proc newContext*(searchDirs = @["./"], partials = initTable[string, string]()): Context =
+  Context(values: initTable[string,Value](), searchDirs: searchDirs, partials: partials)
 
 proc read*(c: Context, filename: string): string =
+  if c.partials.hasKey(filename):
```
Comment (pietroppeter) [inline src/mustachepkg/values.nim:35]: if I read this correctly, partials has precedence over searchDirs. Is this intended?

**Comment by pietroppeter** [comment]
Hi, like the library (thanks @soasme) and also the feature (thanks @iffy). I was thinking of using it and in my use case I would need *in-memory* partials to be last to be used when there are searchDirs (as a sort of default that can be overriden if searchDirs are defined). I am not sure if @iffy really needs them to have precedence over searchDirs (or if in his case they would be empty).

**Comment by soasme** [comment]
Embed partials are not something specified in https://mustache.github.io/, so it's really up to the library's implementation. I'd like to have this feature but with a more flexible interface. We may explicitly specify the order of partial resolvers (table, dir, or even procs) to fix different use cases.

File: src/mustachepkg/values.nim (hunk 1 above)
Comment (iffy) [reply src/mustachepkg/values.nim:35]: Yes, as it is now, memory has precedence over searchDirs.  It was semi-intentional. I run in development using only `searchDirs` and in release build only using in-memory so I didn't ever have to decide which should really get precedence.  I'm fine with what @soasme describes below (explicitly specifying precedence).

**Comment by soasme** [comment]
Code LGTM. Will improve the order in issue #6.

**Pull request merged**